test/kernel/extra_test.exs

```diff
@@ -22,4 +21,35 @@ defmodule Kernel.ExtraTest do
       end
     end
   end
+
+  describe "defbang" do
+    defmodule TestModule do
+      import Kernel.Extra
+      require Integer
+
+      def only_even(x) when Integer.is_even(x), do: {:ok, x}
+      def only_even(_), do: {:error, "not even"}
+
+      def only_odd(x) when Integer.is_odd(x), do: {:ok, x}
+      def only_odd(_), do: :error
+
+      defbang only_even: 1, only_odd: 1
```

Comment:
Stylistically I think it's weird (although admittedly concise) to define multiple functions in one line like this. I would prefer if it forced a single function at a time only. Open to counter examples or logic though.

Reply:
This entire API was based on the API for `defoverridable`, including the ability to call it on multiple functions. More generally, I somewhat see a future world where people don't even really think of functions like `foo` and `foo!` as separate functions, instead thinking of `!` as an "operator" you add to the end of a function name to force it to succeed - to that end, I could imagine a future world where (maybe with the overriding of `defmodule` we're thinking about) "bang" versions of *all* functions get defined automatically, and we don't even really have to think about it as a step in its own.

That's a little ambitious, so even if you disagree with that as a goal my argument about mimicking the API of `defoverridable` still stands.

Reply:
It actually does make more sense given that context and I actually do agree with the overarching goal.

I was pondering why this bothered me, since it was just a feel instead of anything concrete. I think it is from a debugging perspective where I want to find docs or the code for a specific function that I called, it felt like it would be harder to backtrack. `defoverridable` (if im reading it correctly) doesn't generate a new function specifically - it changes an existing function.

BUT THEN thinking the next level down, I think the only real reason why it was bugging me was being able to find specific docs for the function - which is solved by Will's recommendation. The issue of finding the specific code for the function is really not much more difficult than finding any other macro, so I don't think I have any real problems any more.

TL/DR - sounds good to me!